department-of-veterans-affairs/vets-api
# [VI-638] Call MHV::AccountCreatorJob after login

## Summary
- Call `MHV::AccountCreatorJob` after login
- SSOe - called in  `Login::AfterLoginActions`
- SiS -  called in `SignIn::SessionCreator`
- Restricted with a setting for now - `Settings.mhv.account_creation.create_after_login` (default is `false`)

## Related issue(s)
- https://jira.devops.va.gov/browse/VI-638

## Testing
- Test on SiS and SSOe 
  - start `vets-api` and `vets-website`
  - login
  - navigate to http://localhost:3000/sidekiq/queues/default
     - you shouldn't see any `MHV::AccountCreatorJob` enqueues
  - in `settings.local.yml` update mhv.account_creation.create_after_login to true
    ```yml
    mhv:
      account_creation:
        create_after_login: true
    ```
  - restart `vets-api`
  - login
  - navigate to http://localhost:3000/sidekiq/queues/default
     - you should see a `MHV::AccountCreatorJob` enqueued
     <img width="404" alt="Screenshot 2024-10-10 at 3 02 52 PM" src="https://github.com/user-attachments/assets/4ac7e7b4-ddc2-42d0-9e3b-1c3d9c56a592">

  - Repeat for other login method
   
## What areas of the site does it impact?
- Authentication
- MHV Account Creation

## Acceptance criteria
